src/Raven.Server/Documents/ClusterTransactionIndexWaiter.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Sparrow.Server;
+using Sparrow.Utils;
+
+namespace Raven.Server.Documents;
+
+public class ClusterTransactionIndexWaiter
```

Comment:
inherit this from `AbstractRaftIndexNotifications`

Reply:
Wonder if we need to wire this with to the database notification as well

Reply:
> AbstractRaftIndexNotifications

Why do you want to inherit?
Most of the code is not relevant.

> Wonder if we need to wire this with to the database notification as well

It will require a big refactor of the code.